static/app/utils/replays/replayReader.tsx

```diff
@@ -71,6 +89,44 @@ export default class ReplayReader {
     replayRecord,
     errors,
   }: RequiredNotNull<ReplayReaderParams>) {
+    const {breadcrumbFrames, optionFrame, rrwebFrames, spanFrames} =
+      hydrateFrames(attachments);
+
+    if (localStorageWrapper.getItem('REPLAY-BACKEND-TIMESTAMPS') !== '1') {
+      // TODO(replays): We should get correct timestamps from the backend instead
+      // of having to fix them up here.
+      const {startTimestampMs, endTimestampMs} = replayTimestamps(
+        replayRecord,
+        rrwebFrames,
+        breadcrumbFrames,
+        spanFrames
+      );
+      replayRecord.started_at = new Date(startTimestampMs);
+      replayRecord.finished_at = new Date(endTimestampMs);
+      replayRecord.duration = duration(
+        replayRecord.finished_at.getTime() - replayRecord.started_at.getTime()
+      );
+    }
+
+    // Hydrate the data we were given
+    this.replayRecord = replayRecord;
+    this._errors = hydrateErrors(replayRecord, errors);
+    this._rrwebEvents = rrwebFrames;
+    this._breadcrumbFrames = hydrateBreadcrumbs(replayRecord, breadcrumbFrames);
+    this._spanFrames = hydrateSpans(replayRecord, spanFrames);
+    this._optionFrame = optionFrame;
+
+    // Insert extra records to satisfy minimum requirements for the UI
+    this._breadcrumbFrames.push(replayInitBreadcrumb(replayRecord));
+    this._rrwebEvents.push(recordingStartFrame(replayRecord));
+    this._rrwebEvents.push(recordingEndFrame(replayRecord));
+
+    // Sort what needs sorting
+    this._rrwebEvents.sort((a, b) => a.timestamp - b.timestamp);
```

Comment:
Aside from start/end frames, do we know why rrwebEvents would be unsorted? I think it's fine to be on the safe side as I assume the replay would break if it were unsorted, though now I wonder if the replayer handles sorting as well.

If it's only due to start/end frames, we could probably assume that start frame should be at index 0 so we can `array.unshift`

Reply:
good point. I confirmed that although we process segments async on the server, they're always returned sorted, so the `sort()` here is extra.

Reply:
Nope, we need this sort in ReplayReader!

the backend and everything will load the data in order, but on the frontend we're parallelizing all the fetch requests and not keeping track of their order when results come in. Then end result is that we append chunks of data potentially out of order.

That could be fixed, but i'll leave it as a followup.

Reply:
done!

```
    // Sort what needs sorting
    // TODO(replay): We could remove this sort call if useReplayData was more
    // careful about maintaining cursor order when calling setAttachments()
```

Reply:
PR up: https://github.com/getsentry/sentry/pull/50901